pkg/services/annotations/annotationsimpl/composite_store.go

```diff
@@ -0,0 +1,101 @@
+package annotationsimpl
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"sort"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/services/annotations"
+	"github.com/grafana/grafana/pkg/services/annotations/accesscontrol"
+)
+
+// CompositeStore is a read store that combines two or more read stores, and queries all stores in parallel.
+type CompositeStore struct {
+	readers []readStore
+}
+
+func NewCompositeStore(readers ...readStore) *CompositeStore {
+	return &CompositeStore{
+		readers: readers,
+	}
+}
+
+func (c *CompositeStore) Get(ctx context.Context, query *annotations.ItemQuery, accessResources *accesscontrol.AccessResources) ([]*annotations.ItemDTO, error) {
```

Comment:
I think it makes sense to add godocs here to because in Go dependency between struct and interface is weak, and method is public.